doc/guides/V8-api-for-node.md

```diff
@@ -0,0 +1,210 @@
+## V8 API usage for Node.js
+
+v8 docs, not particularly useful:
+- https://v8.dev/docs
+
+Some API info at https://v8.dev/docs/embed
+
+- An isolate is a VM instance with its own heap.
+  Node has one isolate.
+  Can get from `Local<Object>->GetIsolate()`, `context->GetIsolate()`
+
+- `Local<...>`: A local handle is a pointer to an object. All V8 objects are
+  accessed using handles, they are necessary because of the way the V8 garbage
+  collector works.  Local handles can only be allocated on the stack, in a
+  scope, not with new, and will be deleted as stack unwinds, usually with a
+  scope.
+  - XXX don't see us create scopes much in node, why?
+  - XXX locals go into a HandleScope, for auto-dispose? Always, or by default?
```

Comment:
Can you explain this one?

Reply:
There's always a HandleScope somewhere (creating a Local without a HandleScope anywhere is a fatal error) but it's possibly the HandleScope that's way down the stack, around the central call to `uv_run()`.

Locals that end up in that HandleScope are effectively leaked because they're not reclaimed until the event loop terminates.

Reply:
> There's always a HandleScope somewhere (creating a Local without a HandleScope anywhere is a fatal error) but it's possibly the HandleScope that's way down the stack, around the central call to `uv_run()`.

I’m not sure if it’s worth mentioning here, but we have the `SealHandleScope` around the event loop in order to make sure that we don’t leak those handles.

Reply:
Btw, @jasnell requested in https://github.com/nodejs/node/pull/27116 that we explain when to use `SealHandleScope` – I think this would fit in nicely here? Basically, the idea is to make sure that the code inside the `SealHandleScope` always *needs* to explicitly open a `HandleScope` if it does use handles.